icu4j/main/tests/core/src/com/ibm/icu/dev/test/number/NumberFormatterApiTest.java

```diff
@@ -727,6 +727,28 @@ public void unitMeasure() {
                 4.28571,
                 "4 metric tons, 285 kilograms, 710 grams");
 
+        assertFormatSingle(
+                "Mixed Unit (Not Sorted)",
+                "unit/gram-and-kilogram unit-width-full-name",
+                "unit/gram-and-kilogram unit-width-full-name",
+                NumberFormatter.with()
+                        .unit(MeasureUnit.forIdentifier("gram-and-kilogram"))
+                        .unitWidth(UnitWidth.FULL_NAME),
+                new ULocale("en-US"),
+                4.28571,
+                "285.71 grams, 4 kilograms");
+
+        assertFormatSingle(
+                "Mixed Unit (Not Sorted)",
+                "unit/inch-and-yard-and-foot unit-width-full-name",
+                "unit/inch-and-yard-and-foot unit-width-full-name",
+                NumberFormatter.with()
+                        .unit(MeasureUnit.forIdentifier("inch-and-yard-and-foot"))
```

Comment:
I don't find these two test cases in C++.

Reply:
good catch

Reply:
Still missing.

Reply:
Added. And found the C++ code is broken (might be my "fixes" that broke it, don't know, won't specifically invest in trying to find out, will just fix).

Reply:
(I got a: "4,611,170,626,486,031,624 inches, 4 yards, 0 feet" - uninitialized value? (2^62 = 4.611686e+18. Expected result is "1.88556 inches, 4 yards, 1 foot".)

Reply:
Since the Formattable can now be either an int or a double, I've added case statements to be sure the right getter is called (getInt64() vs getDouble()). This resolves the uninitialized value, but as mentioned elsewhere, there's still a double->integer problem.